test/uk/gov/hmrc/softdrinksindustrylevystub/controllers/SubscriptionControllerSpec.scala

```diff
@@ -25,42 +26,344 @@ import play.api.libs.json.Json
 import play.api.test.FakeRequest
 import play.api.test.Helpers._
 import uk.gov.hmrc.play.http.HeaderCarrier
-import uk.gov.hmrc.softdrinksindustrylevystub.models.DesSubmissionResult
+import uk.gov.hmrc.softdrinksindustrylevystub.models.etmp.createsub.{CreateSubscriptionRequest, CreateSubscriptionResponse}
 import uk.gov.hmrc.softdrinksindustrylevystub.services.DesSubmissionService
 
 class SubscriptionControllerSpec extends PlaySpec with MockitoSugar with GuiceOneAppPerSuite with BeforeAndAfterEach {
   val mockDesSubmissionService: DesSubmissionService = mock[DesSubmissionService]
-  val mockHelloWorldController = new SubscriptionController(mockDesSubmissionService)
+  val mockSubscriptionController = new SubscriptionController(mockDesSubmissionService)
+  implicit val hc: HeaderCarrier = new HeaderCarrier
 
   override def beforeEach() {
     reset(mockDesSubmissionService)
   }
 
-  "HelloWorldController" should {
-    "return Status: OK Body: DesSubmissionResult(true) for successful valid submitDesRequest" in {
-      implicit val hc = new HeaderCarrier
+  "SubscriptionController" should {
 
-      val input = """{ "number": 1 }"""
+    "return Status: 404 Body: CreateSubscriptionRequest for a unsuccessful retrieve request" in {
+      val utr = "1097172565"
 
-      when(mockDesSubmissionService.buildResponse()).thenReturn(DesSubmissionResult(true))
-      val response = mockHelloWorldController.submitHello()(FakeRequest("POST", "/des-valid").withBody(Json.parse(input)))
+      when(mockDesSubmissionService.retrieveSubscriptionDetails(utr)).thenReturn(None)
+      val response = mockSubscriptionController.retrieveSubscriptionDetails(utr)(FakeRequest("GET", "/retrieve-subscription-details/"))
+
+      status(response) mustBe NOT_FOUND
+    }
+
+    "return Status: OK Body: CreateSubscriptionRequest for a successful retrieve request" in {
+      val output = Json.parse("""{
```

Comment:
Maybe we could extract the inputs into json files or a resource package object perhaps?